### PR TITLE
Update model to identify upcoming and overdue tasks

### PR DESCRIPTION
**Pull request by burnflare**

### CODE REVIEW

**Comment by burnflare** [comment]
@rachx I made this PR for you. Please update the title to match what it is actually supposed to me. And link to an issue too.

File: src/main/java/seedu/agendum/model/task/Task.java [hunk 1]
```diff
@@ -124,6 +151,39 @@ public int hashCode() {
     }
 
     @Override
+    public int compareTo(Task other) {
+//        int compareCompletionStatus = compareCompletionStatus(other);
+//        if (compareCompletionStatus != 0) {
+//            return compareCompletionStatus;
+//        }
```
Comment (burnflare) [inline src/main/java/seedu/agendum/model/task/Task.java:158]: Commented code necessary?

**Comment by rachx** [comment]
This was a draft for wei guang to work on. need to clean up first

**Comment by burnflare** [comment]
The commits from this branch have been merged into the change UI branch #43. Because of this there's a dependency for this PR/branch to be merged into master first before we can merge the UI branch.

**Pull request closed without merging**